components/automate-ui/src/app/pages/automate-settings/automate-settings.component.spec.ts

```diff
@@ -88,78 +145,121 @@ describe('AutomateSettingsComponent', () => {
-        .controls.missingNodes.value;
+      const infraNestedForms: IngestJob = {
+        name: 'periodic_purge_timeseries',
+        category: 'infra',
+        disabled: false,
+        threshold: '',
+        purge_policies: {
+          elasticsearch: [
+            {
+              name: 'actions',
+              older_than_days: 22, // default is 30, since disabled
+              disabled: true       // is true older than should be null
+            },
+            {
+              name: 'converge-history',
+              older_than_days: 12,
+              disabled: false
+            }
+          ]
+        }
+      };
 
-      expect(missingNodesValues.disable).toEqual(true);
-      expect(missingNodesValues.threshold).toEqual('60');
-      expect(missingNodesValues.unit).toEqual('m');
-    });
+      const complianceForms: IngestJob = {
+        category: 'compliance',
+        name: 'periodic_purge',
+        threshold: '',
+        disabled: true,
+        purge_policies: {
+          elasticsearch: [
+            {
+              name: 'compliance-reports',
+              older_than_days: 105,
+              disabled: false
+            },
+            {
+              name: 'compliance-scans',
+              older_than_days: 92,
+              disabled: false
+            }
+          ]
+        }
+      };
 
-    it('updates the "deleteMissingNodes" form group correctly', () => {
-      component.updateForm(mockJobSchedulerStatus);
+      const clientRunsRemoveData: IngestJob = {
+        category: 'infra',
+        name: 'missing_nodes',
+        disabled : false,
+        threshold : '7d'
+      };
 
-      const deleteMssingNodesValues = component.automateSettingsForm
-        .controls.deleteMissingNodes.value;
+      const clientRunsLabelMissing: IngestJob = {
+        category: 'infra',
+        name: 'missing_nodes_for_deletion',
+        disabled: false,
+        threshold: '14m'
+      };
 
-      expect(deleteMssingNodesValues.disable).toEqual(false);
-      expect(deleteMssingNodesValues.threshold).toEqual('24');
-      expect(deleteMssingNodesValues.unit).toEqual('h');
+      mockJobSchedulerStatus = new JobSchedulerStatus([
+        eventFeedRemoveData,
+        infraNestedForms,
+        clientRunsRemoveData,
+        clientRunsLabelMissing,
+        complianceForms
+      ]);
     });
 
-    it('does not updates the "eventFeed" form group', () => {
-      component.updateForm(mockJobSchedulerStatus);
+    using([
+      ['eventFeedRemoveData', false, 53 ],
+      ['eventFeedServerActions', true, undefined ], // Infra purge_timeseries -> server_actions
+      // ['serviceGroupNoHealthChecks', false, 5 ], // Services not enabled yet
+      // ['serviceGroupRemoveServices', false, 5 ], // Services not enabled yet
+      ['clientRunsRemoveData', false, '7' ], // Infra Remove data
+      ['clientRunsLabelMissing', false, '14' ], // Infra label as missing data
```

Comment:
Either adjust the input like so or adjust the parameter type in the function signature.
```suggestion
      ['clientRunsRemoveData', false, 7 ], // Infra Remove data
      ['clientRunsLabelMissing', false, 14 ], // Infra label as missing data
```

Reply:
In this particular case, that number is a string for the "older" part of the API and a number fo the "newer" parts of the API.  Would you recommend normalizing this in the frontend somewhere?

Reply:
Actually - you sort of answered this down below, as this could be handled in factory functions.  I think I'm all set